src/lib.rs

```diff
@@ -837,6 +837,87 @@ pub enum Format {
     NoSpaces,
 }
 
+/// The representation of a color stop for a `ColorScale`.
+/// The position defines where the color is placed from left (0.0) to right (1.0).
+#[derive(Debug, Clone)]
+struct ColorStop {
+    color: Color,
+    position: Fraction,
+}
+
+/// The representation of a color scale.
+/// The first `ColorStop` (position 0.0) defines the left end color.
+/// The last `ColorStop` (position 1.0) defines the right end color.
+#[derive(Debug, Clone)]
+pub struct ColorScale {
+    color_stops: Vec<ColorStop>,
+}
+
+impl ColorScale {
+    /// Create an empty `ColorScale`.
+    pub fn empty() -> Self {
+        Self {
+            color_stops: Vec::new(),
+        }
+    }
+
+    /// Add a `Color` at the given position.
+    pub fn add_stop(&mut self, color: Color, position: Fraction) -> &mut Self {
+        let color_stop = ColorStop { color, position };
+
+        let next_index = self
+            .color_stops
+            .iter()
+            .position(|c| position.value() <= c.position.value());
+
+        let index = match next_index {
+            Some(0) => 0,
+            Some(next) => next - 1,
+            None => self.color_stops.len(),
+        };
+
+        self.color_stops.insert(index, color_stop);
+
+        self
+    }
+
+    /// Get the color at the given position using the mixing function.
+    ///
+    /// Note:
+    /// - No color is returned if position isn't between two color stops or the `ColorScale` is empty.
+    pub fn sample(
+        &self,
+        position: Fraction,
+        mix: &Box<dyn Fn(&Color, &Color, Fraction) -> Color>,
```

Comment:
I really like that we can pass in a `mix`ing function. It would be great if we could lift the restriction that it has to be a `Box`ed function. I think you could just take a
``` rust
callback: &dyn Fn(&Color, &Color, Fraction) -> Color
```
argument?

This way, you could maybe also simplify the test code below to
```
color_scale.sample(Fraction::from(…), &Color::mix<Lab>)
```
or similar (not tested)

Reply:
I've tried using `Color::mix<C>` to get the mixing function, but this function depends on `self`.

Maybe I can move `get_mixing_function` from `/cli/colorspace.rs` to `/colorspace.rs`. What do you think?

Reply:
Rust allows to call member functions like free functions, see: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f12b413f9b01716e2d2805c62bc64e7d

Reply:
Oh, I didn't know that... much better this way. :satisfied:

Reply:
Somehow this change does not compile for rustc 1.34.0.

I'm looking the compile error to see why.

Reply:
I have bumped the min. required version, don't worry.